Training/Training/intlist.c

```diff
@@ -0,0 +1,290 @@
+#include "intlist.h"
+#include <malloc.h>
+#include<stdbool.h>
+
+// bool variable to flag deleted lists
+bool isDeleted = false;
+
+struct Node {
+	int data;
+	struct Node* next; //we can create pointers to incomplete types
+};
+
+
+/// <summary>
+/// This function declares and initialises a Node pointer to NULL and returns it.
+/// </summary>
+/// <returns>Node Pointer</returns>
+struct Node* Create () {
+	struct Node* head = NULL;
+	isDeleted = false;
+	return head;
+}
+
+
+/// <summary>
+/// This function allocates space for a new Node and assigns values to its members
+/// </summary>
+/// <param name="data"></param>
+/// <returns>Node Pointer</returns>
+struct Node* CreateNode (int data) {
+
+	struct Node* newNode = (struct Node*)malloc (sizeof (struct Node));
+	if (newNode != NULL) {
+		newNode->data = data;
+		newNode->next = NULL;
+	}
+	return newNode;
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// The function then checks if the head pointer points to a list(empty or with nodes).
+/// If it points to a node, the node's address is temporarily stored in a pointer. The head pointer is modified to point to 
+/// the next node and the previous node is freed.
+/// </summary>
+/// <param name="head"></param>
+/// <returns>Error code (if any)</returns>
+int Delete (struct Node** head) {
+
+	if (isDeleted == true) {

```

Comment:
What if there are 2 linked lists and you deleted one? You have a single "isDeleted" flag for multiple lists. How will this work?